### DISCUSSION TITLE
Does Vue InstantSearch support TypeScript?

### DISCUSSION THREAD

**Discussion by hermesalvesbr** [algolia/instantsearch#5821] [Q&A]
![Captura de tela de 2023-08-22 10-56-56](https://github.com/algolia/instantsearch/assets/379850/b107bea4-fd3e-430f-b49e-bab6a03c41d0)

I'm trying to use the vue component, but it alerts lack of types in typescript. I saw that React supports types, doesn't it exist in Vue?

**Comment by Haroenv** [comment]
We don't have types for Vue InstantSearch as it's only supported in Vue 3 and we're using a single codebase to support Vue 2 and Vue 3. If you'd find a suitable way to support it, we'd be happy to have a look

**Reply by hermesalvesbr** [reply]
May you embed this in your package?

https://github.com/algolia/vue-instantsearch/issues/378#issuecomment-783496586

**Reply by Haroenv** [reply]
That's something that would fit better on DefinitelyTyped, which you're free to add @hermesalvesbr

**Comment by MangoMarcus** [comment]
Sorry to piggyback this discussion but has anyone got types working with vue3? I'm using the types mention here: https://github.com/algolia/vue-instantsearch/issues/378#issuecomment-783496586

I've saved them as `vue-instantsearch.d.ts`, and changed the first line to `declare module 'vue-instantsearch/vue3/es'` but still no joy when I use `import { ... } from 'vue-instantsearch/vue3/es'` 🙁
